File: RLMS/data_prep.py
"""
Download the individual-level data for the Russia Longitudinal Monitoring Survey:

https://dataverse.unc.edu/dataset.xhtml?persistentId=doi:10.15139/S3/12438

There are data files and codebook files, you will want one of each.
"""

import numpy as np
import pandas as pd

# Variables to keep
va = [
        "idind", # individual ID
        "J1", # current work status (1 = working)
        "age", # current age
        "year", # current year
        "J69_9C", # year of birth
        "educ", # education
        "status", # area type (city, etc.)
        "J8", # hours worked last 30 days
        "J10", # after tax wages
        "H5", # gender 1 = male, 2 = female
        "psu", # survey PSU and geographic area
        "OCCUP08", # occupation code
     ]

df = pd.read_csv("RLMS-HSE_IND_1994_2018_STATA.tab.gz", sep="\t", usecols=va)

# Drop people who are not working
df = df.loc[df.J1==1, :]

# Recode gender
df["Female"] = (df.H5 == 2).astype(np.int)

# Drop the original versions of variables that we no longer need.
df = df.drop(columns=["J1", "H5"])

# Drop rows with missing values
dx = df.dropna()

# Center year at 2000, making it more centered
dx.year -= 2000

# Remove special codes
for v in "J10", "J8", "educ", "age":
    ii = dx.loc[:, v] < 99999997
    dx = dx.loc[ii, :]

File: RLMS/data_prep.R
library(dplyr)
library(readr)

# Variables to keep
#va = cols_only(idind, "J1", "age", "year", "J69_9C", "educ",
#      "status", "J8", "J10", "H5", "psu", "OCCUP08")

c = cols_only(idind=col_double(), J1=col_double(), age=col_double(), year=col_double(),
              J69_9C=col_double(), educ=col_double(), status=col_factor(), J8=col_double(),
              J10=col_double(), H5=col_double(), psu=col_factor(), OCCUP08=col_factor())

df = read_tsv("RLMS-HSE_IND_1994_2018_STATA.tab.gz", col_types=c)

# Drop people who are not working
df = df[df$J1==1,]

# Recode gender
df$female = as.integer(df$H5 == 2)

# Drop the original versions of variables that we no longer need.
df = subset(df, select=-c(J1, H5))

# Drop rows with missing values
dx = df[complete.cases(df),]

# Center year at 2000, making it more centered
dx$year = dx$year - 2000

# Remove special codes
for (v in c("J10", "J8", "educ", "age")) {
    ii = dx[[v]] < 99999997
    dx = dx[ii,]
}